astropy/table/table.py

```diff
@@ -1753,6 +1757,7 @@ def _make_index_row_display_table(self, index_row_name):
         else:
             return self
 
+    @deprecated("6.1", pending=True)
```

Comment:
Was there a conversation on why you opt to go with PendingDeprecationWarning instead of a full DeprecationWarning? This does not officially kick off the deprecation period AFAIK.

Also change log is missing.

Reply:
Nope, not yet. I just put in a `PendingDeprecationWarning` as I wasn't sure if there is enough consensus in deprecating it. I can change this now with the change log entry.

Reply:
It seems the consensus is to go ahead with deprecation now on both methods.